proofs/tree.go

```diff
@@ -384,13 +408,36 @@ func getSaltsFromMessage(message proto.Message) (salts []*proofspb.Salt, err err
 	return field.Interface().([]*proofspb.Salt), nil
 }
 
+func emptyNodeHash(h hash.Hash) ([]byte, error) {
+	defer h.Reset()
+	_, err := h.Write([]byte{})
+	if err != nil {
+		return []byte{}, err
+	}
+	hash := h.Sum(nil)
+	return hash, nil
+}
+
 // Generate calculated the merkle root with all supplied leaves. This method can only be called once and makes
 // the tree immutable.
 func (doctree *DocumentTree) Generate() error {
 	if doctree.filled {
 		return errors.New("tree already filled")
 	}
 
+	if doctree.fixedNoOfLeafs != 0 {
+		emptyNoToBeAdded := doctree.fixedNoOfLeafs - uint(len(doctree.leaves))
+		if emptyNoToBeAdded > 0 {
+			hash, err := emptyNodeHash(doctree.hash)
```

Comment:
This will cause an issue when we merge #80 (PR #81) Adding a comment here so we don't forget this.

Reply:
@touhonoob Would you be able to address this? I just merged this pull request. It should use the leaf node hash function here

Reply:
@lucasvo I'll have a look

Reply:
fixed in https://github.com/centrifuge/precise-proofs/pull/81/commits/4700de7049e357fcb20dd0fdddba0677c0c530b3